prog/mutation.go

```diff
@@ -395,49 +458,125 @@ type mutationArgs struct {
 	target        *Target
 	args          []Arg
 	ctxes         []ArgCtx
+	priorities    []float64
 	ignoreSpecial bool
 }
 
+const (
+	maxPriority = float64(10)
+	minPriority = float64(1)
+)
+
 func (ma *mutationArgs) collectArg(arg Arg, ctx *ArgCtx) {
+	var prio float64
+	var canMutate bool
+
 	ignoreSpecial := ma.ignoreSpecial
 	ma.ignoreSpecial = false
+
 	switch typ := arg.Type().(type) {
-	case *StructType:
+	case *StructType, *UnionType:
 		if ma.target.SpecialTypes[typ.Name()] == nil || ignoreSpecial {
-			return // For structs only individual fields are updated.
-		}
-		// These special structs are mutated as a whole.
-		ctx.Stop = true
-	case *UnionType:
-		if ma.target.SpecialTypes[typ.Name()] == nil && len(typ.Fields) == 1 || ignoreSpecial {
 			return
 		}
-		ctx.Stop = true
-	case *ArrayType:
-		// Don't mutate fixed-size arrays.
-		if typ.Kind == ArrayRangeLen && typ.RangeBegin == typ.RangeEnd {
-			return
-		}
-	case *CsumType:
-		return // Checksum is updated when the checksummed data changes.
-	case *ConstType:
-		return // Well, this is const.
-	case *BufferType:
-		if typ.Kind == BufferString && len(typ.Values) == 1 {
-			return // string const
-		}
 	case *PtrType:
```

Comment:
What happens if we fall into this case (or the previous one) and the `if` check on the following line fails? We won't call `getMutationPrio()`, right?

Reply:
Yes. I should remove the default case.